plugins/Files/js/components/filelist.js

```diff
@@ -0,0 +1,68 @@
+import React, { PropTypes } from 'react'
+import { List } from 'immutable'
+import File from './file.js'
+import Directory from './directory.js'
+import Path from 'path'
+import SearchField from '../containers/searchfield.js'
+
+const FileList = ({files, searchResults, path, showSearchField, actions}) => {
+
+	const onDirectoryClick = (directory) => () => actions.setPath(path + directory.name + '/')
+	const onBackClick = () => {
+		if (path === '') {
+			return
+		}
+		let newpath = Path.join(path, '../')
+		if (newpath === './') {
+			newpath = ''
+		}
+		actions.setPath(newpath)
+	}
+
+	let filelistFiles
+	if (showSearchField) {
+		filelistFiles = searchResults
+	} else {
+		filelistFiles = files
+	}
```

Comment:
does it make sense to combine these in a single component? Maybe. Alternatively, search could be its own component that wraps `FileList`, providing it with the files returned by the search flattened into a single directory.